### ISSUE TITLE
Anatomy of shield style

### ISSUE THREAD

**Issue by wipfli** [ZeLonewolf/maplibre-shield-rotation-sample#6]
Based on #5, I conclude that the shield style currently looks like this:

```json
        {
            "id": "road_label",
            "type": "symbol",
            "paint": {
                "text-color": "#333",
                "text-halo-color": "#fff",
                "text-halo-blur": 0.5,
                "text-halo-width": 1
            },
            "filter": [
                "all",
                [
                    "==",
                    "class",
                    "motorway"
                ]
            ],
            "layout": {
                "text-font": [
                    "Metropolis Light"
                ],
                "text-size": 20,
                "text-field": [
                    "format",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_1"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_1"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_1"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ],
                    " ",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_2"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_2"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_2"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ],
                    " ",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_3"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_3"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_3"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ],
                    " ",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_4"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_4"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_4"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ],
                    " ",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_5"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_5"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_5"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ],
                    " ",
                    [
                        "case",
                        [
                            "!=",
                            [
                                "get",
                                "ref_6"
                            ],
                            null
                        ],
                        [
                            "image",
                            [
                                "concat",
                                "shield_",
                                [
                                    "get",
                                    "network_5"
                                ],
                                "=",
                                [
                                    "get",
                                    "ref_6"
                                ]
                            ]
                        ],
                        [
                            "literal",
                            ""
                        ]
                    ]
                ],
                "text-anchor": "center",
                "symbol-placement": "line"
            },
            "source": "openmaptiles",
            "metadata": {},
            "source-layer": "transportation_ref"
        }
```

**Comment by wipfli** [comment]
So it uses the source layer `transportation_ref`. Do you think we could reduce the number of cases to have a simpler example? If we can figure out how to rotate for one case, then the others will be trivial. What would be a good case to use?

**Comment by wipfli** [comment]
Ah, and if I could pick one, I would want to be able to render the I70 shields :)

**Comment by ZeLonewolf** [comment]
Sure, we could just make a style fragment that hard-codes the same two-shield combination for every case.  Is that the kind of thing you're thinking of?

**Comment by wipfli** [comment]
Yes. I think we need a minimal example which just shows a diagonal highway with a tilted shield. All the other stuff, like water and tunnels etc, should be removed.

**Comment by ZeLonewolf** [comment]
Besides removing features we don't need, which is straightforward,  do we also need simpler tiles? Or is what we have ok?

**Comment by wipfli** [comment]
Tiles are fine I think.

**Comment by ZeLonewolf** [comment]
Resolved by #8
